### PR TITLE
[Meteor] - Add types for MongoInternals

### PR DESCRIPTION
**Pull request by harryadel**
Added types for MongoInternals which got lifted from https://github.com/Favro/meteor-typescript/blob/master/definitions/meteor.d.ts#L123 Thanks @Favro.

### CODE REVIEW

File: types/meteor/mongo.d.ts [hunk 1]
```diff
@@ -219,6 +240,8 @@ declare module "meteor/mongo" {
             map<M>(callback: (doc: U, index: number, cursor: Cursor<T, U>) => M, thisArg?: any): Array<M>;
             observe(callbacks: ObserveCallbacks<U>): Meteor.LiveQueryHandle;
             observeChanges(callbacks: ObserveChangesCallbacks<T>, options?: { nonMutatingCallbacks?: boolean }): Meteor.LiveQueryHandle;
+            _mongo: MongoInternals.Connection;
+            _synchronousCursor: MongoInternals.SynchronousCursor;
```
Comment (radekmie) [inline types/meteor/mongo.d.ts:244]: Can't we use the `mongodb` package here?

Comment (harryadel) [reply types/meteor/mongo.d.ts:244]: Can you please elaborate? My sole purpose of adding those types is such code block would yield no errors:

```
const connection = await MongoClient.connect(
      MongoInternals.defaultRemoteCollectionDriver().mongo.client.s.url,
      {
        useNewUrlParser: true,
        useUnifiedTopology: true,
        poolSize: 1,
      }
    );

    const db = await connection.db(
      MongoInternals.defaultRemoteCollectionDriver().mongo.client.s.options
        .dbName
    );
```

Comment (radekmie) [reply types/meteor/mongo.d.ts:245]: Of course. What I meant was to not type missing properties ad hoc, but rather use typings of the `mongodb` package (imported at the top of the file) instead.

Comment (harryadel) [reply types/meteor/mongo.d.ts]: Thanks for your response <3. Please bear with me.
`mongodb` package lacks explicit `Connection` type. The closest thing we have to it is `Collection` interface given both have `find` and `findOne` properties but even then, it still lacks important keys like `mongo`, `db`, `_oplogHandle` and `_createSynchronousCursor`. I presume these properties are added by [meteor](https://github.com/meteor/meteor/blob/46e00a875726dff078c59eaf7fdc7b956cf0a996/packages/mongo/mongo_driver.js#L201) which makes it unpractical to have them exported from `mongodb`

But as for `synchronousCursor`, `Mongo.Cursor` may be a sufficient alternative. Please double check with me.

Comment (radekmie) [reply types/meteor/mongo.d.ts]: Indeed, a few of these methods, like [`_createSynchronousCursor` are indeed Meteor-specific](https://github.com/meteor/meteor/blob/46e00a875726dff078c59eaf7fdc7b956cf0a996/packages/mongo/mongo_driver.js#L982). But I don't like the idea of typing _such_ internal code. Wouldn't `mongo` and `db` (created [here](https://github.com/meteor/meteor/blob/46e00a875726dff078c59eaf7fdc7b956cf0a996/packages/mongo/mongo_driver.js#L249-L251)) be enough? You can always `await` asynchronous methods as well as use `for await of` for cursors. Does it fit your needs?

Comment (harryadel) [reply types/meteor/mongo.d.ts]: Thanks for the great tip! The previous code block just got shortened to:
```
    const { db } = MongoInternals.defaultRemoteCollectionDriver().mongo;
```
 with no need to use `mongodb` npm driver package.
 
 But still I've 
 ```
 Module '"meteor/mongo"' has no exported member 'MongoInternals'.
 ```
 Since I'm importing `MongoInternals`
 
 ```
 import { MongoInternals } from 'meteor/mongo';
```

So, I'll see what we cut from the PR.

File: types/meteor/README.md [hunk 2]
```diff
@@ -5,18 +5,15 @@
 These are the definitions for version 1.4 of Meteor.  These definitions were generated from the same [Meteor data.js file](https://github.com/meteor/meteor/blob/devel/docs/client/data.js) that is used to generate the official [Meteor docs](http://docs.meteor.com/).  The code that generates these definitions can be found [here](https://github.com/meteor-typescript/meteor-typescript-libs/).
 
 
-## Upcoming Meteor `typescript` package
+## Meteor `typescript` package
 
-There is currently an effort supported by the Meteor Development Group to create a TypeScript build compiler package, and an early version of the package can be tested using [`barbatus:typescript`](https://atmospherejs.com/barbatus/typescript).
+Meteor currently offers an official [typescript package](https://atmospherejs.com/meteor/typescript).
```
Comment (radekmie) [inline types/meteor/README.md:10]: Could you merge in the `master` branch? This should be already there.

File: types/meteor/mongo.d.ts [hunk 3]
```diff
@@ -219,6 +241,8 @@ declare module "meteor/mongo" {
             map<M>(callback: (doc: U, index: number, cursor: Cursor<T, U>) => M, thisArg?: any): Array<M>;
             observe(callbacks: ObserveCallbacks<U>): Meteor.LiveQueryHandle;
             observeChanges(callbacks: ObserveChangesCallbacks<T>, options?: { nonMutatingCallbacks?: boolean }): Meteor.LiveQueryHandle;
+            _mongo: MongoInternals.Connection;
+            _synchronousCursor: Mongo.Cursor<Mongo.Document>;
```
Comment (radekmie) [inline types/meteor/mongo.d.ts:245]: Since we agreed that these are not needed, I'd remove it completely, along with the code added above.

File: types/meteor/mongo.d.ts [hunk 4]
```diff
@@ -243,3 +267,15 @@ declare module "meteor/mongo" {
         }
     }
 }
+
+declare module MongoInternals {
+    interface Connection {
```
Comment (radekmie) [inline types/meteor/mongo.d.ts:272]: To match [the source](https://github.com/meteor/meteor/blob/46e00a875726dff078c59eaf7fdc7b956cf0a996/packages/mongo/mongo_driver.js#L136):
```suggestion
    interface MongoConnection {
```

File: types/meteor/mongo.d.ts [hunk 5]
```diff
@@ -243,3 +267,15 @@ declare module "meteor/mongo" {
...
+        db: MongoDb;
```
Comment (radekmie) [inline types/meteor/mongo.d.ts:273]: Could we also add `client` here?

**Review by radekmie — CHANGES_REQUESTED** [review]
Hopefully the last review :rocket:

File: types/meteor/mongo.d.ts (hunk 5 above)
Comment (harryadel) [reply types/meteor/mongo.d.ts:273]: What'd I set it to? `MongoDb` also?

Comment (radekmie) [reply types/meteor/mongo.d.ts:271]: No, [`MongoClient`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/882f5542a4f0f9793f944f3650adec7e7ee98fe3/types/mongodb/index.d.ts#L51-L53).

File: types/meteor/mongo.d.ts [hunk 6]
```diff
@@ -1,10 +1,32 @@
...
+
+        interface FindOptions<T> extends FindOneOptions<T> {
+            limit?: number;
+        }
+
+        interface CursorDescription<T> {
+            collectionName: string;
+            selector: any;
+            options: FindOptions<T>;
+        }
```
Comment (radekmie) [inline types/meteor/mongo.d.ts:29]: These are no longer needed, right?

Comment (harryadel) [reply types/meteor/mongo.d.ts]: Great catch! I forgot adding them :sweat_smile:

File: types/meteor-astronomy/meteor-astronomy-tests.ts [hunk 7]
```diff
@@ -79,6 +79,7 @@ interface UserInterface extends Meteor.User {
     phone: string;
     phoneNumber: string;
     fullName: (param: string) => string;
+    profile: UserProfileInterface;
```
Comment (radekmie) [inline types/meteor-astronomy/meteor-astronomy-tests.ts:82]: Please remove this unrelated change.

Comment (harryadel) [reply types/meteor-astronomy/meteor-astronomy-tests.ts:82]: :( But the test were failing without this change!

Comment (radekmie) [reply types/meteor-astronomy/meteor-astronomy-tests.ts:82]: That's weird... OK.

**Review by radekmie — APPROVED** [review]

**Comment by harryadel** [comment]
Thank you @radekmie for your time and help. <3

**Comment by harryadel** [comment]
Thank you @radekmie for your time and help. <3

**Comment by harryadel** [comment]
Ready to merge
